packages/components/src/components/date-picker/date-picker.tsx

```diff
@@ -130,17 +149,17 @@ const DatePicker = React.memo(props => {
     /**
      * TODO: currently only works for duration, make it works for date as well
      */
-    const onChangeInput = e => {
-        const new_date = addDays(toMoment(), e.target.value).format(display_format);
+    const onChangeInput: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = e => {
+        const new_date = addDays(toMoment(), isNaN(Number(e.target.value)) ? 0 : Number(e.target.value)).format(
+            display_format
+        );
         const new_duration = mode === 'duration' ? e.target.value : '';
 
         setDate(new_date);
         setDuration(new_duration);
         setIsDatepickerVisible(true);
         setIsPlaceholderVisible(false);
 
-        calendar_ref.current?.setSelectedDate(new_date);
```

Comment:
please elaborate why do we remove it and what we're replacing it with?🙂

Reply:
@maryia-deriv actually the `ref` is an `HTMLDivElement` and there is no such function we can call from a `div` element `ref` , typescript was also complaining.

I cannot see a possibility of this function call that's why I removed it

Reply:
yeah, makes sense😄 I think it will work anyway like I assumed above and anyway we'll have this checked by QA👍🏻 nice work!